src/devices/bus/mc10/mc10_cart.cpp

```diff
@@ -108,6 +108,18 @@ std::pair<std::error_condition, std::string> mc10cart_slot_device::call_load()
 				image_error::INVALIDLENGTH,
 				util::string_format("Unsupported cartridge size (must be no more than %u bytes)", m_cart->max_rom_length()));
 	}
+	else if (len < m_cart->min_rom_length())
+	{
+		return std::make_pair(
+				image_error::INVALIDLENGTH,
+				util::string_format("Unsupported cartridge size (must be at least %u bytes)", m_cart->min_rom_length()));
+	}
+	else if (len % m_cart->block_rom_length() != 0)
+	{
+		return std::make_pair(
+				image_error::INVALIDLENGTH,
+				util::string_format("Unsupported cartridge size (must be a multiple of %u bytes)", m_cart->block_rom_length()));
+	}
```

Comment:
Adding the minimum length and granularity to the interface over-complicates things when they can be checked in `load()` if necessary anyway.  The only reason the maximum length is checked here is to guard against attempting to read an excessively large file as loose software.

Reply:
Will do.

Reply:
Change done